qutebrowser/javascript/webelem.js

```diff
@@ -188,26 +278,56 @@ window._qutebrowser.webelem = (function() {
             return null;
         }
 
-        return serialize_elem(elem);
+        // Check if we got an iframe, and if so, recurse inside of it
+        const frame_elem = replace_elem_frame(elem,
+            (frame) => serialize_elem(frame.document.activeElement, frame));
+
+        if (frame_elem === null) {
+            return serialize_elem(elem);
+        }
+        return frame_elem;
     };
 
     funcs.find_at_pos = (x, y) => {
-        // FIXME:qtwebengine
-        // If the element at the specified point belongs to another document
-        // (for example, an iframe's subdocument), the subdocument's parent
-        // element is returned (the iframe itself).
-
         const elem = document.elementFromPoint(x, y);
-        return serialize_elem(elem);
+
+
+        // Check if we got an iframe, and if so, recurse inside of it
+        const frame_elem = replace_elem_frame(elem,
+            (frame) => {
+                // Subtract offsets due to being in an iframe
+                const frame_offset_rect =
+                      frame.frameElement.getBoundingClientRect();
+                return serialize_elem(frame.document.
+                    elementFromPoint(x - frame_offset_rect.left,
+                        y - frame_offset_rect.top), frame);
+            });
+
+        if (frame_elem === null) {
+            return serialize_elem(elem);
+        }
+        return frame_elem;
     };
 
     // Function for returning a selection to python (so we can click it)
     funcs.find_selected_link = () => {
         const elem = window.getSelection().anchorNode;
-        if (!elem) {
+        if (elem) {
+            return serialize_elem(elem.parentNode);
+        }
+
+        const serialized_frame_elem = run_frames((frame) => {
+            const node = frame.window.getSelection().anchorNode;
+            if (node) {
+                return serialize_elem(node.parentNode, frame);
+            }
             return null;
+        });
+
+        if (serialized_frame_elem) {
+            return serialized_frame_elem;
```

Comment:
This is `null` anyways when nothing was found, right? So you should be able to just `return serialized_elem` unconditionally here.